lieer/local.py

```diff
@@ -632,52 +622,40 @@ def update_tags (self, m, fname, db):
       else:
         print ("(dry-run) tried to update tags on non-existant file: %s" % fname)
 
-    nmsg  = db.find_message_by_filename (fname)
+    try:
+      nmsg = db.get(fname)
+    except LookupError:
+      nmsg = None
 
     if nmsg is None:
       if self.dry_run:
         print ("(dry-run) adding message: %s: %s, with tags: %s" % (gid, fname, str(labels)))
       else:
         try:
-          if hasattr (notmuch.Database, 'index_file'):
-            (nmsg, _) = db.index_file (fname, True)
-          else:
-            (nmsg, _) = db.add_message (fname, True)
-        except notmuch.errors.FileNotEmailError:
+          (nmsg, _) = db.add(fname, sync_flags = True)
+        except notmuch2.errors.FileNotEmailError:
           print('%s is not an email' % fname)
           return True
-        nmsg.freeze ()
-
-        # adding initial tags
-        for t in labels:
-          nmsg.add_tag (t, True)
-
-        for t in self.new_tags:
-          nmsg.add_tag (t, True)
+        with nmsg.frozen():
+          # adding initial tags
+          nmsg.tags |= labels
+          nmsg.tags |= self.new_tags
```

Comment:
do we not need tags to maildir flag sync anymore?

Reply:
I could not find any function for this in notmuch2 python interfaces.  It was my assumption that after the tags are changed, and if the notmuch config has 'synchronize_flags=true', then this would happen automatically.  Though, I'm not very familiar with what is going on behind the scenes here.

Reply:
I missed this.  So I checked the new API and you need to call `nmsg.tags.to_maildir_flags()`.  The call is within a method of the `MutableTagSet` instance.  I can update the PR (if I understand how github PRs go).

Reply:
Thanks, pushing a new commit to the branch you made your PR from will add it to the PR.

Reply:
Looks good!